.github/workflows/ci.yml

```diff
@@ -44,11 +44,12 @@ jobs:
         id: composer-cache
         run: |
           echo "::set-output name=dir::$(composer config cache-files-dir)"
-      - uses: actions/cache@v3
+
+      - name: Cache composer dependencies
+        uses: actions/cache@v3
         with:
           path: ${{ steps.composer-cache.outputs.dir }}
-          key: composer-php${{ matrix.php }}-typo3${{ matrix.typo3 }}
-
+          key: composer-${{ hashFiles('composer.json') }}-php${{ matrix.php }}-typo3${{ matrix.typo3 }}
```

Comment:
This project does not contain a composer.json file and does not cache the related /vendor folder.
Instead it caches composer's internal cache directory (based on TYPO3 and PHP version)

Reply:
Don't we want to have a fresh cache when we raise the dependencies in our extension `composer.json`? Thats why I use the file hash of the `composer.json` as we do not have a `composer.lock` file. On the other side composer can handle this for use, but then the cache can increase a lot, but I don't think that we hit the limit of 10GB.

Reply:
Seems like I mixed up composer.lock and composer.json :facepalm: 

Using composer.json results in a stale cache key that will not change for months (probably not until TYPO3 12 is added as dependency) and should not any benefit in controlling cache bahaviour.

I just checked composer cache on my local machine (after running clear-cache, then install). It uses ~15M of disk space. It will take a long time until the official limit of 10G is maxed out.

Anyway, according to [official GitHub docs](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy), caches are purged if they are not accessed in over 7 days. I suppose this extension will have more time between builds, therefore caches would be empty either way.